crates/bevy_ecs/macros/src/component.rs

```diff
@@ -47,19 +54,36 @@ pub fn derive_entity_event(input: TokenStream) -> TokenStream {
         .predicates
         .push(parse_quote! { Self: Send + Sync + 'static });
 
-    for attr in ast.attrs.iter().filter(|attr| attr.path().is_ident(EVENT)) {
+    for attr in ast
+        .attrs
+        .iter()
+        .filter(|attr| attr.path().is_ident(ENTITY_EVENT))
+    {
         if let Err(e) = attr.parse_nested_meta(|meta| match meta.path.get_ident() {
             Some(ident) if processed_attrs.iter().any(|i| ident == i) => {
                 Err(meta.error(format!("duplicate attribute: {ident}")))
             }
             Some(ident) if ident == AUTO_PROPAGATE => {
+                propagate = true;
                 auto_propagate = true;
                 processed_attrs.push(AUTO_PROPAGATE);
                 Ok(())
             }
             Some(ident) if ident == TRAVERSAL => {
-                traversal = meta.value()?.parse()?;
-                processed_attrs.push(TRAVERSAL);
+                Err(meta.error(
+                    "`traversal` has been renamed to `propagate`, use that instead. If you were writing `traversal = &'static ChildOf`, you can now just write `propagate`, which defaults to the ChildOf traversal."
```

Comment:
I like the change to cause this to default. Very user-friendly, even if it is a bit more magical.